ksql-engine/src/test/java/io/confluent/ksql/ddl/commands/CommandFactoriesTest.java

```diff
@@ -825,6 +830,115 @@ public void shouldFailDropSourceOnDropIncompatibleSourceForStream() {
     commandFactories.create("sqlExpression", dropStream, ksqlConfig, emptyMap());
   }
 
+  @Test
+  public void shouldThrowOnRowTimeValueColumn() {
+    // Given:
+    final DdlStatement statement = new CreateStream(
+        SOME_NAME,
+        TableElements.of(tableElement(Namespace.VALUE, ROWTIME_NAME, new Type(SqlTypes.BIGINT))),
+        true,
+        withProperties
+    );
+
+    // Then:
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage("'ROWTIME' is a reserved field name.");
+
+    // When:
+    commandFactories.create("sqlExpresson", statement, ksqlConfig, emptyMap());
+  }
+
+  @Test
+  public void shouldThrowOnRowTimeKeyColumn() {
+    // Given:
+    final DdlStatement statement = new CreateStream(
+        SOME_NAME,
+        TableElements.of(tableElement(Namespace.KEY, ROWTIME_NAME, new Type(SqlTypes.BIGINT))),
+        true,
+        withProperties
+    );
+
+    // Then:
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage("'ROWTIME' is a reserved field name.");
+
+    // When:
+    commandFactories.create("sqlExpresson", statement, ksqlConfig, emptyMap());
+  }
+
+  @Test
+  public void shouldThrowOnRowKeyValueColumn() {
+    // Given:
+    final DdlStatement statement = new CreateStream(
+        SOME_NAME,
+        TableElements.of(tableElement(VALUE, ROWKEY_NAME, new Type(SqlTypes.STRING))),
+        true,
+        withProperties
+    );
+
+    // Then:
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage(
+        "'ROWKEY' is a reserved field name. It can only be used for KEY fields.");
+
+    // When:
+    commandFactories.create("sqlExpresson", statement, ksqlConfig, emptyMap());
+  }
+
+  @Test
+  public void shouldNotThrowOnRowKeyKeyColumn() {
+    // Given:
+    final DdlStatement statement = new CreateStream(
+        SOME_NAME,
+        TableElements.of(tableElement(KEY, ROWKEY_NAME, new Type(SqlTypes.STRING))),
+        true,
+        withProperties
+    );
+
+    // When:
+    commandFactories.create("sqlExpresson", statement, ksqlConfig, emptyMap());
+
+    // Then: did not throw
```

Comment:
If there is a bug in commandFactories.create so it does absolutely nothing this test would still pass. Can we verify somehow that it did what we want it to do?

Reply:
You've picked up on this in other PRs as well.  I disagree with your reasoning.  The point of this test is to prove it does not throw an exception under a certain circumstance.  Yes, an empty implementation, or one with a bug, might pass this test.  But there should be other tests that cover that and they should fail.  This test is only testing that no exception is thrown. It is a valid test.